src/plugins/ipmi_intf.c

```diff
@@ -128,17 +128,40 @@ struct ipmi_intf * ipmi_intf_table[] = {
 	NULL
 };
 
+/* get_default_interface - return the interface that was chosen by configure
+ *
+ * returns a valid interface pointer
+ */
+static struct ipmi_intf *get_default_interface(void)
+{
+	static const char *default_intf_name = DEFAULT_INTF;
+	struct ipmi_intf ** intf;
+	for (intf = ipmi_intf_table; intf && *intf; intf++) {
+		if (strncmp(default_intf_name, (*intf)->name,
+		            strlen(default_intf_name)) == 0)
+		{
+			return *intf;
+		}
+	}
+	/* code should never reach this because the configure script checks
+	 * to see that the default interface is actually enabled, but we have
+	 * to return some valid value here, so the first entry works
+	 */
+	return ipmi_intf_table[0];
+}
+
 /* ipmi_intf_print  -  Print list of interfaces
  *
  * no meaningful return code
  */
 void ipmi_intf_print(struct ipmi_intf_support * intflist)
```

Comment:
Funny thing is that this `intflist` is never actually used and there is no hint as to why it has been introduced in the first place back in 2005 by Duncan...

Reply:
I noticed that too. Maybe it was part of a half-baked dynamically loaded interface mechanism.